### ISSUE TITLE
Problem with cirillic 

### ISSUE THREAD

**Issue by esliklientovnet** [jburkhardt/RAdwords#60]
I have problem this report contain cirillic letters
in function 
transformData
function (data, report = reportType, apiVersion = "201609") 
{
    data <- read.csv2(textConnection(data), sep = ",", header = F)[-1, ...

textConnection have problem this cirillic letter "Я" - data broken after this letter
Please, try change to
data <- read.csv2(text=data, sep = ",", header = F)[-1, ...

**Comment by jburkhardt** [comment]
Thanks for the note. There seems to be an encoding issue with R and your operating system.

Could you provide the following information please:

- Operating System
- locale settings
- R Version
- RAdwords version
- the R code which leads to the error
- the error message, prompt output


What do geht with:
```
getOption("encoding")
```

Further readings on encoding in R:
https://developer.r-project.org/Encodings_and_R.html

**Comment by esliklientovnet** [comment]
R version 3.3.3 (2017-03-06)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)
locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    
other attached packages:
[1] RAdwords_0.1.11

in locale English_United States.1252 i receive escaped unicode like <U+0441><U+0430><U+043C><U+0430><U+044F>....
if i change locale to Russian Sys.setlocale(locale = "Russian") i have error:
incomplete final line found by readTableHeader on 'data'
  or data cut after first letter "Я"

if set Sys.setlocale("LC_ALL","Russian_Russia.20866") i receive all data and see corect in table View() RStudio, but in console i not see right words.

getOption("encoding")
[1] "native.enc"
try change to options(encoding = "UTF-8") - not work

truble in textConnection:
> readLines(textConnection (c("test","тест","я проверяю")),3)
[1] "test" "тест"
> readLines(textConnection (c("test","я проверяю","тест")),3)
[1] "test"
data after leter "я" ignored

**Comment by jburkhardt** [comment]
@mbannert Ironically R has trouble with the cyrillic backwards R. Do you have a clue how we could overcome encoding settings in russian locale system settings?

**Comment by jburkhardt** [comment]
https://github.com/jburkhardt/RAdwords/issues/62

**Comment by googlogmob** [comment]
**It seems there are encoding problems with cyrillic letters correlated with the locale of your operating system. Use Sys.getlocale() to gather information about the locale of your different opertaing systems**

Really, VS and RStudio have different LC_CTYPE
http://prnt.sc/ehf8rx
maybe it is possible to set manualy this encoding to LC_CTYPE=Russian_Russia.866 ?

**Comment by jburkhardt** [comment]
You can use ```Sys.setlocale()``` to change th locale settings on your local machine. Further readings:
http://stackoverflow.com/questions/20577764/set-locale-to-system-default-utf-8

**Comment by googlogmob** [comment]
**You can use Sys.setlocale() to change th locale settings on your local machine. Further readings:
http://stackoverflow.com/questions/20577764/set-locale-to-system-default-utf-8**

I executed this code
`Sys.getlocale()
Sys.setlocale(category = "LC_CTYPE", locale = "Russian_Russia.866")
Sys.getlocale()`
http://prntscr.com/ehff2m


Yes, it works for rows count 
but now all cilillic symbols in text fields is incorrect
http://prntscr.com/ehffr3

But now all cyrillics symbols in text fiels is incorrect

**Comment by esliklientovnet** [comment]
I agree -  error in R and 2 byte locale, but with error we can round, if not use textConnection
Demo:
Sys.setlocale(locale = "Russian")
[1] "LC_COLLATE=Russian_Russia.1251;LC_CTYPE=Russian_Russia.1251;LC_MONETARY=Russian_Russia.1251;LC_NUMERIC=C;LC_TIME=Russian_Russia.1251"
> cyr <-c("test","тест","я проверяю")
> read.csv2(text=cyr)
        test
1       тест
2 я проверяю
#correct
> read.csv2(textConnection (cyr))
     test
1    тест
2  **провер** 
#last not correct

Please, change function transformData : data <- read.csv2(textConnection(data), sep = ",", header = F)[-1, ..., 
and all locale will be work
change locale - wrong way - all other text print incorrect

**Comment by jburkhardt** [comment]
I changed ```textConnection(data)```to ```text=data``` as suggested.

Please reinstall the package from my github repo with:
```
require(devtools)
install_github('jburkhardt/RAdwords')
```
Is the output fine now or do you still have problems?

**Comment by esliklientovnet** [comment]
YES! Many thanks! And best wishes! All work in Russian locale!

**Comment by jburkhardt** [comment]
@googlogmob Is your problem solves as well?

**Comment by googlogmob** [comment]
@jburkhardt  Thank You
Now works well!
@esliklientovnet  Спасибо за решение проблемы

**Comment by jburkhardt** [comment]
Спасибо вам, ребята

**Comment by mbannert** [comment]
@jburkhardt I don't know why ever used text connection. w.r.t the note you dropped I don't think we should experience any trouble here. Plus, if there any problems users will report them anyway. No problems with an US locale, German and Swiss locale I guess. I think the error was caused by forcing an encoding through `textConnection`'s default to the encoding argument. If I only could remember why we used textConnection :)
